cli/cmd/install.go

```diff
@@ -695,6 +702,16 @@ func (options *installOptions) buildValuesWithoutIdentity(configs *pb.All) (*cha
 	installValues.ProxyInit.IgnoreInboundPorts = strings.Join(inboundPortStrs, ",")
 	installValues.ProxyInit.IgnoreOutboundPorts = strings.Join(outboundPortStrs, ",")
 
+	if options.controlPlaneTrace {
```

Comment:
I find this block of code very confusing since we do a mutation of `installValues.ControlPlaneTrace.ProxyTrace.CollectorSvcAddr` to append the namespace and port.  And we do a few more relative mutations which are hard to understand.

Does the `installValues.ControlPlaneTrace.ProxyTrace.CollectorSvcAccount` not end with a namespace?  Does `installValues.Proxy.Trace.CollectorSvcAddr` end with a namespace?

What might be simpler would be to change the chart to something like:

```
- -trace-collector=linkerd-collector.{{.Namespace}}.svc.{{.ClusterDomain}}:55678
```

and have the `ControlPlaneTrace` field in Values.go simply be a boolean.  Does that seem reasonable?

Reply:
I think what's happening here is that we are trying to copy the tracing config from the control plane components to the tracing env vars of the control plane proxies.

I agree that since this is mainly used for control plane debugging, we have liberty over how the config is structured. So something like this makes sense to me:

```
# values.yaml
ControlPlaneTracing: false

# charts/partials/templates/_tracing.tpl
{{ define "partials.linkerd.tracing" -}}
- -trace-collector=linkerd-collector.{{.Namespace}}.svc.{{.ClusterDomain}}:55678
{{- end }}

# charts/partials/templates/_proxy.tpl
# if the whitespaces started to cause malformed YAML, try use `helm template` to debug or DM me
{{ if .ControlPlaneTracing -}}
- name: LINKERD2_PROXY_TRACE_COLLECTOR_SVC_ADDR
  value: linkerd-collector.{{.Namespace}}.svc.{{.ClusterDomain}}:55678
- name: LINKERD2_PROXY_TRACE_COLLECTOR_SVC_NAME
  value: linkerd-collector.{{.Namespace}}.serviceaccount.identity.$(_l5d_ns).$(_l5d_trustdomain)
{{ else if .Proxy.Trace -}}
{{ if .Proxy.Trace.CollectorSvcAddr -}}
- name: LINKERD2_PROXY_TRACE_COLLECTOR_SVC_ADDR
  value: {{ .Proxy.Trace.CollectorSvcAddr }}
- name: LINKERD2_PROXY_TRACE_COLLECTOR_SVC_NAME
  value: {{ .Proxy.Trace.CollectorSvcAccount }}.serviceaccount.identity.$(_l5d_ns).$(_l5d_trustdomain)
{{ end -}}
{{ end -}}
```